pkg/acctest/test.go

```diff
@@ -0,0 +1,160 @@
+	}
+}
+
+// PreCheck This function should be present in every acceptance test.
+func PreCheck(t *testing.T) {
+	// Since we are outside the scope of the Terraform configuration we must
+	// call Configure() to properly initialize the provider configuration.
+	testAccProviderConfigure.Do(func() {
+		restyClient := GetTestResty(t)
+
+		artifactoryUrl := GetArtifactoryUrl(t)
+		// Set custom base URL so repos that relies on it will work
+		// https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateCustomURLBase
+		_, err := restyClient.R().
+			SetBody(artifactoryUrl).
+			SetHeader("Content-Type", "text/plain").
+			Put("/artifactory/api/system/configuration/baseUrl")
+		if err != nil {
+			t.Fatalf("Failed to set custom base URL: %v", err)
+		}
+
+		configErr := Provider.Configure(context.Background(), terraform.NewResourceConfigRaw(nil))
+		if configErr != nil {
+			t.Fatalf("Failed to configure provider %v", configErr)
+		}
+	})
+}
+
+func GetArtifactoryUrl(t *testing.T) string {
+	return test.GetEnvVarWithFallback(t, "PIPELINES_URL", "JFROG_URL")
+}
+
+func GetTestResty(t *testing.T) *resty.Client {
+	artifactoryUrl := GetArtifactoryUrl(t)
+	restyClient, err := client.Build(artifactoryUrl, "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	accessToken := test.GetEnvVarWithFallback(t, "PIPELINES_ACCESS_TOKEN", "JFROG_ACCESS_TOKEN")
+	restyClient, err = client.AddAuth(restyClient, "", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return restyClient
+}
+
+func CreateProject(t *testing.T, projectKey string) {
+	type AdminPrivileges struct {
+		ManageMembers   bool `json:"manage_members"`
+		ManageResources bool `json:"manage_resources"`
+		IndexResources  bool `json:"index_resources"`
+	}
+
+	type Project struct {
+		Key             string          `json:"project_key"`
+		DisplayName     string          `json:"display_name"`
+		Description     string          `json:"description"`
+		AdminPrivileges AdminPrivileges `json:"admin_privileges"`
+	}
+
+	restyClient := GetTestResty(t)
+
+	project := Project{
+		Key:         projectKey,
+		DisplayName: projectKey,
+		Description: fmt.Sprintf("%s description", projectKey),
+		AdminPrivileges: AdminPrivileges{
+			ManageMembers:   true,
+			ManageResources: true,
+			IndexResources:  true,
+		},
+	}
+
+	_, err := restyClient.R().
+		SetBody(project).
+		Post("/access/api/v1/projects")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func DeleteProject(t *testing.T, projectKey string) {
+	restyClient := GetTestResty(t)
+	_, err := restyClient.R().Delete("/access/api/v1/projects/" + projectKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func CreateProjectIntegration(t *testing.T, name string, projectKey string) int {
+	restyClient := GetTestResty(t)
+
+	project := pipeline.ProjectIntegration{
+		Name: name,
+		Project: pipeline.ProjectJSON{
+			Name: projectKey,
+			Key:  projectKey,
+		},
+		MasterIntegrationId:   78,
```

Comment:
Where is this magic number from?

Reply:
The Master Get Integrations API: https://www.jfrog.com/confluence/display/JFROG/Pipelines+REST+API#PipelinesRESTAPI-masterIntegrationsGetallIntegrations